es6-shim/es6-shim.d.ts

```diff
@@ -526,7 +526,7 @@ interface PromiseConstructor {
      * @param reason The reason the promise was rejected.
      * @returns A new rejected Promise.
      */
-    reject(reason: any): Promise<void>;
+    reject(reason: any): Promise<any>;
```

Comment:
I think I am ok with this, though you are always able to use the generic overload below to specify the type.  In the next version of the compiler you will be able to specify the `undefined` type as opposed to `void`. I am wary of using `any` because it will turn off type checking and intellisense for the promised value depending on how it is used.